docs/transactions.md

```diff
@@ -0,0 +1,93 @@
+
+A nested write lets you perform a single Photon API call with multiple _operations_ that touch multiple [_related_](./relations.md#nested-writes) records, for example creating a _user_ together with a _post_ or updating an _order_ together with an _invoice_. When a nested write is performed, Photon ensures that it will either succeed or fail as a whole.
+
+Here are examples for nested writes in the Photon API:
+
+```ts
+// Create a new user with two posts in a 
+// single transaction
+const newUser: User = await photon.users.create({
+  data: {
+    email: 'alice@prisma.io',
+    posts: {
+      create: [
+        { title: 'Join the Prisma Slack on https://slack.prisma.io' },
+        { title: 'Follow @prisma on Twitter' },
+      ],
+    },
+  },
+})
+```
+
+```ts
+// Change the author of a post in a single transaction
+const updatedPost: Post = await photon.posts.update({
+  where: { id: 42 },
+  data: {
+    author: {
+      connect: { email: 'alice@prisma.io' },
+    },
+  },
+})
+```
+
+## Future transaction support in the Photon API
+
+Transactions are a commonly used feature in relational as well as non-relational databases and Photon might support more transaction mechanisms in the future. Specifically, the following two use cases will be supported:
+
+- Sending multiple operations in bulk.
+- Enabling longer-running transactions where operations can depend on each other.
+
+The first use use case of sending multiple operations in bulk could be implemented with an API similar to this:
+
+```ts
+const write1 = photon.users.create()
+const write2 = photon.orders.create()
+const write3 = photon.invoices.create()
+
+await prisma.transaction([write1, write2, write3])
+```
+
+Instead of immediately awaiting the result of each operation when it's performed, the operation itself is stored in a variable first which later is submitted to the database via a method called `transaction`. Photon will ensure that either all three `create`-operations or none of them succeed.
+
+The second use case of longer-running transactions where operations can depend on each other is a bit more involved. Photon would need to expose a _transaction API_ which enables developers to initiate and commit a transaction themselves while Photon takes care of ensuring the safety guarantees associated with transactions. It could look similar to this:
+
+```ts
+prisma.transaction(async tx => {
+  const user = await tx.users.create({
+    data: { email: "alice@prisma.io" }
+  })
+  const order = await tx.orders.create({
+    data: {
+      customer: {
+        connect: { id: user.id }
+      }
+    }
+  })
+  await tx.commit()
+})
+```
+
+In this case, the API provides a way to wrap a sequence of operations in a callback which gets executed as a transaction, therefore is guaranteed to either succeed or fail as a whole.
+
+If you'd like to see transactions supported in the future, [please join the discussion on GitHub](https://github.com/prisma/prisma/issues/4155).
```

Comment:
That link is super weird - shouldn't there be something in prisma2 or specs?

Reply:
Thanks for catching this, yeah this must have been a mistake. I updated the link to an issue in the `prisma2` repo.